core/unit_test/TestMathematicalSpecialFunctions.hpp

```diff
@@ -1710,11 +1710,12 @@ struct TestComplexBesselH1Function {
     h_ref_ch11(24) =
         Kokkos::complex<double>(-5.430453818237824e-02, -1.530182458039000e-02);
 
-    // FIXME_HIP Disable the test when using ROCm 5.5 and 5.6 due to a known
-    // compiler bug
-#if !defined(KOKKOS_ENABLE_HIP) || (HIP_VERSION_MAJOR != 5) || \
-    ((HIP_VERSION_MAJOR == 5) &&                               \
-     !((HIP_VERSION_MINOR == 5) || (HIP_VERSION_MINOR == 6)))
+    // FIXME_HIP Disable the test when using ROCm 5.5, 5.6, and 6.2 due to a
+    // known compiler bug
+#if !defined(KOKKOS_ENABLE_HIP) ||                              \
+    (!((HIP_VERSION_MAJOR == 5) && (HIP_VERSION_MINOR == 5)) && \
+     !((HIP_VERSION_MAJOR == 5) && (HIP_VERSION_MINOR == 6)) && \
+     !((HIP_VERSION_MAJOR == 6) && (HIP_VERSION_MINOR == 2)))
```

Comment:
Maybe this would be slightly easier to read:
```suggestion
#if !(defined(KOKKOS_ENABLE_HIP) &&                          \
      ((HIP_VERSION_MAJOR == 5 && HIP_VERSION_MINOR == 5) || \
       (HIP_VERSION_MAJOR == 5 && HIP_VERSION_MINOR == 6) || \
       (HIP_VERSION_MAJOR == 6 && HIP_VERSION_MINOR == 2)))
```